### PR TITLE
Add game model

### PR DESCRIPTION
**Pull request by obulkin**
- Generated a game model
- Built out and ran the associated migration
- Added validations to the model
- Created a base factory for the model
- Wrote tests for the model logic and any associated DB constraints

### CODE REVIEW

File: app/models/game.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Game < ActiveRecord::Base
+  validates :name, presence: true
+  validates :white_player_id, presence: true
+  validates :black_player_id, presence: true
```
Comment (lasley) [inline app/models/game.rb]: We will run into issues down the road with these being required due to needing to join a game. There's not going to be many instances in which a game will be created immediately with both players. Maybe just a required on whichever the first player is?

File: spec/models/game_spec.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
+require 'rails_helper'
+
+RSpec.describe Game, type: :model do
```
Comment (lasley) [inline spec/models/game_spec.rb:3]: Excellent tests!

File: app/models/game.rb (hunk 1 above)
Comment (obulkin) [inline app/models/game.rb]: Good point. I'll replace these with a custom validation that looks for at least one ID and will remove the constraints on the columns.

Comment (lasley) [inline app/models/game.rb]: :+1:

**Pull request merged**